src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+            this->animInfoIndex = ENSGORO_ANIM_IDLE_STAND;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+        }
+
+        this->scaleFactor = 0.01f;
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actor.gravity = -1.0f;
+        this->actor.flags |= ACTOR_FLAG_10;
+        this->actor.flags |= ACTOR_FLAG_2000000;
+        this->actor.targetMode = 1;
+
+        switch (EN_S_GORO_GET_MAIN_TYPE(&this->actor)) {
+            case ENSGORO_TYPE_SHRINE_WINTER_A:
+                this->actionFunc = EnSGoro_ActionWinterShrineGoron;
+                return;
+            case ENSGORO_TYPE_SHRINE_WINTER_B:
+                this->actionFunc = EnSGoro_ActionWinterShrineGoron;
+                return;
+            case ENSGORO_TYPE_SHRINE_WINTER_C:
+                this->actionFunc = EnSGoro_ActionWinterShrineGoron;
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_A:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_B:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_C:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_D:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_E:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_SHRINE_SPRING_F:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_ActionSpringShrineGoron;
+                    return;
+                }
+                return;
+            case ENSGORO_TYPE_BOMBSHOP:
+                this->actionFlags |= ENSGORO_ACTIONFLAG_ROLLEDUP;
+                this->actionFunc = EnSGoro_ActionShopGoron;
+                this->actor.shape.yOffset = EN_S_GORO_ROLLEDUP_YOFFSET;
+                return;
+            default:
+                Actor_MarkForDeath(&this->actor);
+                return;
+        }
+    }
+}
+
+void EnSGoro_ActionWinterShrineGoron(EnSGoro* this, PlayState* play) {
```

Comment:
For these action functions, we don't prefix them with `Action`. Instead the functions take the form of some verb. This one in particular looks like the Winter Shrine Goron's Idle function (they are just waiting around for the player to interact with them). So could be called something like `EnSGoro_WinterShrineGoron_Idle`, note this gets split as `WinterShrineGoron` seems to be a subsystem/subtype of EnSGoro.